### DISCUSSION TITLE
Backend fails on AWS deployment

### DISCUSSION THREAD

**Discussion by fgimenez** [blockscout/blockscout#9157] [Support] [answered]
I have configured blockscout to run on an Elastic Container Service on AWS with the docker images to connect to an OP Stack based L2. I've followed geth's docker compose setup to configure everything, using `blockscout/blockscout-optimism:6.0.0-postrelease-294b304f` image for the backend and the rest as defined in the docker compose files. The initial migration works fine but when the backend starts, I get the following error:

```
{"time":"2024-01-15T14:51:31.272Z","severity":"error","message":"GenServer Indexer.Fetcher.OptimismDeposit terminating\n** (FunctionClauseError) no function clause matching in Indexer.Fetcher.OptimismDeposit.terminate/2\n    (indexer 6.0.0) lib/indexer/fetcher/optimism_deposit.ex:332: Indexer.Fetcher.OptimismDeposit.terminate(:normal, %{})\n    (stdlib 3.17) gen_server.erl:733: :gen_server.try_terminate/3\n    (stdlib 3.17) gen_server.erl:918: :gen_server.terminate/10\n    (stdlib 3.17) proc_lib.erl:226: :proc_lib.init_p_do_apply/3\nLast message: {:continue, :ok}\nState: %{}","metadata":{"error":{"initial_call":null,"reason":"** (FunctionClauseError) no function clause matching in Indexer.Fetcher.OptimismDeposit.terminate/2\n    (indexer 6.0.0) lib/indexer/fetcher/optimism_deposit.ex:332: Indexer.Fetcher.OptimismDeposit.terminate(:normal, %{})\n    (stdlib 3.17) gen_server.erl:733: :gen_server.try_terminate/3\n    (stdlib 3.17) gen_server.erl:918: :gen_server.terminate/10\n    (stdlib 3.17) proc_lib.erl:226: :proc_lib.init_p_do_apply/3\n"},"fetcher":"optimism_deposits"}}
```

and at the end of the log:
```
=INFO REPORT==== 15-Jan-2024::14:51:43.060159 ===
    application: logger
    exited: shutdown
    type: permanent

{"Kernel pid terminated",application_controller,"{application_terminated,logger,shutdown}"}
Kernel pid terminated (application_controller) ({application_terminated,logger,shutdown})

Crash dump is being written to: /scripts/erl_crash.dump...done

```

If I run locally docker compose with the same common-blockscout.env file everything works fine.

Any help is very appreciated.

common-blockscout.env file: https://gist.github.com/fgimenez/c36c58fdace5a2f9169067c7f6cc72fb
ECS task definition: https://gist.github.com/fgimenez/14a1a75a9451a09712d763fd7901d724
Full backend log: https://gist.github.com/fgimenez/4a95f9fc299efdca57ce7550e4cdde52

**Comment by ekawibisono** [comment]
hello @fgimenez Have you tried with the default version of blockscout which is tags/v6.0.0?

**Comment by fgimenez** [comment, marked as answer]
hey @ekawibisono it finally worked by increasing `INDEXER_MEMORY_LIMIT` to 5gb, the rest of optimism settings the same, thanks for your help!
